storage/client/queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1,29 +1,479 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 package com.azure.storage.queue;
 
-import com.azure.core.ServiceClient;
 import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.http.rest.VoidResponse;
+import com.azure.core.util.Context;
+import com.azure.storage.common.credentials.SASTokenCredential;
+import com.azure.storage.common.credentials.SharedKeyCredential;
 import com.azure.storage.queue.implementation.AzureQueueStorageBuilder;
 import com.azure.storage.queue.implementation.AzureQueueStorageImpl;
-
+import com.azure.storage.queue.models.DequeuedMessage;
+import com.azure.storage.queue.models.EnqueuedMessage;
+import com.azure.storage.queue.models.MessageIdUpdateHeaders;
+import com.azure.storage.queue.models.MessageIdsUpdateResponse;
+import com.azure.storage.queue.models.PeekedMessage;
+import com.azure.storage.queue.models.QueueGetPropertiesHeaders;
+import com.azure.storage.queue.models.QueueMessage;
+import com.azure.storage.queue.models.QueueProperties;
+import com.azure.storage.queue.models.QueuesGetPropertiesResponse;
+import com.azure.storage.queue.models.SignedIdentifier;
+import com.azure.storage.queue.models.StorageErrorException;
+import com.azure.storage.queue.models.UpdatedMessage;
+import java.net.MalformedURLException;
 import java.net.URL;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * This class provides a client that contains all the operations for interacting with a queue in Azure Storage Queue.
+ * Operations allowed by the client are creating and deleting the queue, retrieving and updating metadata and access
+ * policies of the queue, and enqueuing, dequeuing, peeking, updating, and deleting messages.
+ *
+ * <p><strong>Instantiating an Asynchronous Queue Client</strong></p>
+ *
+ * <pre>
+ * QueueAsyncClient client = QueueAsyncClient.builder()
+ *     .connectionString(connectionString)
+ *     .endpoint(endpoint)
+ *     .queueName(queueName)
+ *     .buildAsyncClient();
+ * </pre>
+ *
+ * <p>View {@link QueueClientBuilder this} for additional ways to construct the client.</p>
+ *
+ * @see QueueClientBuilder
+ * @see QueueClient
+ * @see SharedKeyCredential
+ * @see SASTokenCredential
+ */
+public final class QueueAsyncClient {
+    private final AzureQueueStorageImpl client;
+    private final String queueName;
+
+    /**
+     * Creates a QueueAsyncClient that sends requests to the storage queue service at {@link AzureQueueStorageImpl#url() endpoint}.
```

Comment:
This is not a public method, so it is not visible to users. However, please validate all JavaDoc for issues. For example, here AzureQueueStorageImpl is not public API, so it should not be referred to in JavaDoc.

Reply:
Will remove the link here. And review for the rest.